### PR TITLE
Refactor/config commands

### PR DESCRIPTION
**Pull request by LhKipp**
The config commands used the variable "config-path". Now they use `ctx.configs` :)
I also removed all other occurrences of "config-path".

Edit: Oops, I duplicated some of the config tests. I didn't saw them . Can I move the existing config tests (which tests the config commands, and only those) to nu-command/tests/command/config.rs?

### CODE REVIEW

File: crates/nu-engine/src/evaluate/variables.rs [hunk 1]
```diff
@@ -17,18 +21,33 @@ pub fn nu(scope: &Scope, tag: impl Into<Tag>) -> Result<Value, ShellError> {
             dict.insert_untagged(v.0, UntaggedValue::string(v.1));
         }
     }
+
     nu_dict.insert_value("env", dict.into_value());
 
-    let config_file = match scope.get_var("config-path") {
```
Comment (fdncred) [inline crates/nu-engine/src/evaluate/variables.rs:22]: Please help me understand why are we removing `$nu.config-path`?

Comment (LhKipp) [reply crates/nu-engine/src/evaluate/variables.rs:22]: Running: `echo $nu.config-path` will still give the expected output :)
It's right on the other side:  https://github.com/nushell/nushell/pull/3265/files#diff-c8858d9cba0d39d1870b2e7169ae3c176590b812be484adf52f62c076eb2ac89R38-R41

Comment (fdncred) [reply crates/nu-engine/src/evaluate/variables.rs:22]: lolz. thanks!

File: crates/nu-engine/src/evaluation_context.rs [hunk 2]
```diff
@@ -188,30 +188,8 @@ impl EvaluationContext {
     /// If an error occurs while reloading the config:
     ///     The config is not reloaded
     ///     The error is returned
-    pub fn reload_config(&self, cfg_path: &ConfigPath) -> Result<(), ShellError> {
```
Comment (LhKipp) [inline crates/nu-engine/src/evaluation_context.rs:191]: We now pass NuConfig instead of Path, so the next 20 lines error handling is obsolete

**Comment by jonathandturner** [comment]
Cool - looks good

**Pull request merged**